core/modules/field/field.block.inc

```diff
@@ -76,6 +86,12 @@ class FieldBlock extends Block {
     list($entity_type, $field_name) = explode('-', $this->childDelta);
     $entity = $this->contexts[$entity_type]->data;
 
+    // Do not render if the entity is not found on this layout.
+    if (empty($entity)) {
+      watchdog('field', 'Attempted to display a field block from a missing entity.', array(), WATCHDOG_WARNING);
```

Comment:
IMO we shouldn't log here. But if we did, we should at least provide information on how to solve the problem, such as linking to the Layout UI configuration page, so that an admin knows where to go to fix it.